packages/gatsby-dev-cli/src/watch.js

```diff
@@ -27,23 +27,24 @@ const copyPath = (oldPath, newPath, quiet) =>
   })
 
 function watch(root, packages, { scanOnce, quiet }) {
-  const ignored = [
-    /[/\\]node_modules[/\\]/i,
-    /\.git/i,
-    /\.DS_Store/,
-  ].concat(
+  const ignored = [/[/\\]node_modules[/\\]/i, /\.git/i, /\.DS_Store/].concat(
     packages.map(p => new RegExp(`${p}[\\/\\\\]src[\\/\\\\]`, `i`))
   )
   const watchers = packages.map(p => path.join(root, `/packages/`, p))
 
   let allCopies = []
 
-  chokidar.watch(watchers, {
-    ignored: [filePath => _.some(ignored, reg => reg.test(filePath))],
-  })
+  chokidar
+    .watch(watchers, {
+      ignored: [filePath => _.some(ignored, reg => reg.test(filePath))],
+    })
     .on(`all`, (event, filePath) => {
-      if (event === `change` || event === `add`) {
-        const packageName = path.basename(path.dirname(filePath.split(`packages/`).pop()))
```

Comment:
Given something like `packages/gatsby/dist/gatsby/cache-dir/something.js` this was inferring the `packageName` to be `cache-dir` rather than gatsby. This is the main fix here.